File: tests/arch/fixtures/real_world/csharp/src/Result/ResultFactory.cs
using Example.Result.IResult;
using Example.Result.Success;
using Example.Result.Failure;

namespace Example.Result;

/// <summary>Factory helpers for creating typed result instances.</summary>
public static class ResultFactory
{
    public static IResult<T> Ok<T>(T value) => new Success<T>(value);
    public static IResult<T> Fail<T>(string error) => new Failure<T>(error);
}

File: tests/arch/fixtures/real_world/php/src/Result/ResultFactory.php
<?php

declare(strict_types=1);

namespace Example\Result;

use Example\Result\IResult;
use Example\Result\Success;
use Example\Result\Failure;

/**
 * Factory helpers for creating typed result instances.
 */
final class ResultFactory
{
    /** @param mixed $value */
    public static function ok(mixed $value): IResult
    {
        return new Success($value);
    }

    public static function fail(string $error): IResult
    {
        return new Failure($error);
    }
}
